net/Socket.hpp

```diff
@@ -186,13 +186,17 @@ class Socket
     virtual void shutdown()
     {
         if (_noShutdown)
+        {
+            setClosed();
```

Comment:
Why don't we do setClosed at the top of this function just once (?)

Reply:
good point, fixed